warehouse/query-core/src/test/java/datawave/query/data/parsers/EventKeyTest.java

```diff
@@ -59,6 +61,17 @@ public void testRepeatedParsing() {
 
         parser.parse(eventKeyWithChildUidNoDashes);
         assertKeyWithChildUidNoDashes(parser);
+
+        // this should skip actual processing, since it was already parsed up once.
+        Key dupeKey = new Key("row", "datatype\0d8zay2.3pnndm.anolok.12.34", "FIELD\0value");
+        parser.parse(dupeKey);
+        assertKeyWithChildUidNoDashes(parser);
+        assertNotNull(parser.getDatatype());
```

Comment:
Can we assert the values instead of just asserting "not null"? You can shove them in some variables and reuse.

Reply:
yep, will fix this.